### PR TITLE
refactor(Task): removing gotoTask field

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

In the `StatefulTask` interface we had `gotoTask` and `action`, the first one was just a kinda simplification of the second. This PR remove the `gotoTask`, rename `action` to `rendererAction` and deprecate it.

> ℹ️ The goal of those are to remove the rendererAction, to have all nicely be done in the main.

### Screenshot / video of UI

**before**
image

**After**

image

### What issues does this PR fix or reference?

Related to https://github.com/containers/podman-desktop/issues/7401

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by deboer-tim — COMMENTED** [review]
👍🏼 to the cleanup, but why rename? Especially if you're deprecating I don't know why to bother.

I also don't understand what would come next after the deprecation, as this isn't linked to a specific issue/clear what the replacement would be.

Based on before and after images, my understanding is that you've made this 10s faster. ;-)

**Review by jeffmaury — COMMENTED** [review]
Renaming something that you are deprecating seems opposite

File: packages/api/src/task.ts
Comment (axel7083) [inline packages/api/src/task.ts]:
```suggestion
  action?: {
```

File: packages/renderer/src/lib/container/ContainerExport.svelte
Comment (axel7083) [inline packages/renderer/src/lib/container/ContainerExport.svelte]:
```suggestion
  task.action= {
```

File: packages/renderer/src/lib/image/build-image-task.ts
Comment (axel7083) [inline packages/renderer/src/lib/image/build-image-task.ts]:
```suggestion
  task.action= {
```

File: packages/renderer/src/lib/preferences/preferences-connection-rendering-task.ts
Comment (axel7083) [inline packages/renderer/src/lib/preferences/preferences-connection-rendering-task.ts]:
```suggestion
  task.action= {
```

File: packages/renderer/src/lib/task-manager/TaskManagerItem.spec.ts
Comment (axel7083) [inline packages/renderer/src/lib/task-manager/TaskManagerItem.spec.ts]:
```suggestion
  action: {
```

File: packages/renderer/src/lib/task-manager/TaskManagerItem.svelte
Comment (axel7083) [inline packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]:
```suggestion
  taskUI?.action?.execute();
```

File: packages/renderer/src/lib/task-manager/TaskManagerItem.svelte
Comment (axel7083) [inline packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]:
```suggestion
          {#if taskUI.action}
```

File: packages/renderer/src/lib/task-manager/TaskManagerItem.svelte
Comment (axel7083) [inline packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]:
```suggestion
              aria-label="action button">{taskUI.action.name}</button>
```

File: packages/renderer/src/lib/task-manager/task-manager.spec.ts
Comment (axel7083) [inline packages/renderer/src/lib/task-manager/task-manager.spec.ts]:
```suggestion
  action: {
```

File: packages/renderer/src/lib/task-manager/task-manager.ts
Comment (axel7083) [inline packages/renderer/src/lib/task-manager/task-manager.ts]:
```suggestion
        action: task.action,
```

**Comment by axel7083** [comment]
> 👍🏼 to the cleanup, but why rename? Especially if you're deprecating I don't know why to bother.

The idea was the task can have an action, but not on the renderer side

**Comment by jeffmaury** [comment]
Code LGTM but deprecation should come once there is an alternative

**Review by jeffmaury — APPROVED** [review]
Code LGTM but deprecation should come once there is an alternative ?

**Comment by axel7083** [comment]
> Code LGTM but deprecation should come once there is an alternative ?

Sorry I forgot to respond to the first message, the alternative is still under discussion here https://github.com/containers/podman-desktop/issues/7401#issuecomment-2205584869

**Review by deboer-tim — APPROVED** [review]
> > Code LGTM but deprecation should come once there is an alternative ?
> 
> Sorry I forgot to respond to the first message, the alternative is still under discussion here [#7401 (comment)](https://github.com/containers/podman-desktop/issues/7401#issuecomment-2205584869)

I'd prefer you remove the deprecation then until we have an alternative, but code LGTM.

File: packages/api/src/task.ts
Comment (axel7083) [inline packages/api/src/task.ts]:
```suggestion
```

**Comment by axel7083** [comment]
> I'd prefer you remove the deprecation then until we have an alternative, but code LGTM.

Done, if I see anyone using it, just because it is not deprecated I will block their PR!!

**Pull request merged**
